src/contracts/KeyperPayer.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract KeyperPayer is Ownable {
+    using SafeERC20 for IERC20;
+
+    mapping(address => uint) private balances;
+    address[] private keypers;
+    address public sptTokenAddress;
+
+    constructor(address tokenAddress) Ownable(msg.sender) {
+        sptTokenAddress = tokenAddress;
+    }
+
+    function SetKeypers(address[] calldata addresses) public onlyOwner {
+        keypers = addresses;
+    }
+
+    function GetKeypers() public view returns (address[] memory) {
+        return keypers;
+    }
+
+    function Pay(uint amount) external {
+        require(keypers.length > 0, "keypers should be set");
+        uint allowance = IERC20(sptTokenAddress).allowance(
+            msg.sender,
+            address(this)
+        );
+        require(allowance >= amount, "insufficient allowance");
+        IERC20(sptTokenAddress).safeTransferFrom(
+            msg.sender,
+            address(this),
+            amount
+        );
+        uint share = amount / keypers.length;
```

Comment:
@fredo will know this better, but I think that you could run into potential issues with this.

If the amount is not properly divisible by keepers.length you would probably lose the remainder - basically you'll lose tokens?

Reply:
nice catch!, i will have a requirement regarding this.